helix-view/src/handlers/lsp.rs

```diff
@@ -47,6 +47,10 @@ pub enum SignatureHelpEvent {
     RequestComplete { open: bool },
 }
 
+pub enum PullDiagnosticsEvent {
+    Trigger,
+}
```

Comment:
Instead let's make this a struct (for now - we can always refactor as an enum later if we want to support multiple kinds of events) which carries the `DocumentId` from `event.doc.id()` on the `DocumentDidChange` event. Then in the hook you can get the document with `doc!(editor, &doc_id)` or `doc_mut!(editor, &doc_id)` for the mutable version

Reply:
Nice. I added this, but it has resulted in document_id being copied and passed around. 
But it did not make sense to pass a mutable reference to document.